routes/web/webhook/github.rb

```diff
@@ -75,10 +75,14 @@ def handle_workflow_job(data)
       return success("GithubRunner[#{runner.ubid}] created")
     end
 
+    unless (runner_id = job.fetch("runner_id"))
+      return error("A workflow_job without runner_id")
+    end
+
     runner = GithubRunner.first(
       installation_id: installation.id,
       repository_name: data["repository"]["full_name"],
-      runner_id: job.fetch("runner_id")
+      runner_id: runner_id
     )
```

Comment:
Something doesn't add up about this: `job.fetch` already crashes/terminates execution early, and so does `return error`. While it's more graceful to have an HTTP error than a 500 from a `KeyError`, it doesn't explain the bug: in both cases, the route is aborted. It seems to me that a defect as you describe would remain both before and after patch.

```ruby
[4] clover-development(main)> {}.fetch('nope')
KeyError: key not found: "nope"
from (pry):3:in `fetch'
```

Reply:
It has `runner_id` key with a nil value in all cases. `.fetch` isn't failing.

https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=waiting#workflow_job

<img width="435" alt="Screenshot 2023-12-11 at 15 42 47" src="https://github.com/ubicloud/ubicloud/assets/993199/5cce867d-b5be-4e0f-9a51-cbf0bfb17282">

Reply:
Ahh that makes sense.